Cockpit 317 (#733)
cockpit-project/cockpit-project.github.io: _posts/2024-05-28-cockpit-317.md

---
title: Cockpit 317
author: jelle
date: '2024-05-28'
tags: cockpit, machines, files
slug: cockpit-317
category: release

> summary: ''

Reviewer: ```suggestion
summary: 'Switch from static to dynamic system users, Grafana setup now prefers Valkey instead of Redis'
```

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 317:

> ## webserver: System user changes
> 
> This release drops the `./configure` options for customizing the two system users of cockpit's web server:
> `--with-cockpit-user`, `--with-cockpit-group`, `--with-cockpit-ws-instance-user`, and `--with-cockpit-ws-instance-user`. All supported distributions use the same name anyway. We don't test, support, or recommend running the web server as root, which so far has been the default with `./configure; make; make install`.
> 
> The `cockpit-ws` system user is not created statically at all, but created transiently and on demand via [systemd's `DynamicUser` feature](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=). If you want to, you can safely remove the static system user with e.g. `userdel -r cockpit-ws` after the upgrade to this version and after `systemctl stop cockpit`.
> 
> The `cockpit-wsinstance` system user is now declared through [systemd-sysusers](https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html#).

Reviewer: I rewrote the entire block to change the focus and make it a little more obvious what's changing and why.

```suggestion
## Webserver: System user changes

Cockpit has changed how the system user is handled. All supported distributions now use the same system user names.

We don't test, support, or recommend running the web server as root, which had previously been the default when running `./configure; make; make install`.

### Dynamic user

Previously-used Cockpit static users (`cockpit-ws` and `cockpit-wsinstance`) have been replaced by dynamic users.

The `cockpit-ws` system user is no longer statically created, but created transiently and on-demand via [systemd's `DynamicUser` feature](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=).

If you would like to remove the previously-used static system user:

1. Upgrade to Cockpit 317 or later
2. Run `systemctl stop cockpit`
3. Run `userdel -r cockpit-ws`

The `cockpit-wsinstance` system user is now declared through [systemd-sysusers](https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html#).

### Configuration removals

As Cockpit has switched to using dynamic users, the following `./configure` options related to static users have been removed:

- `--with-cockpit-user`
- `--with-cockpit-group`
- `--with-cockpit-ws-instance-user`
- `--with-cockpit-ws-instance-user`
```

Reviewer: It's weird though; we only suggest removing cockpit-ws and not cockpit-wsinstance. Is this intentional, or should we combine the removals for both?

Reviewer 2: Yes, it's very intentional. cockpit-wsinstance *must* be static for now, it's hard to remove. This *only* applies to cockpit-ws (which is DynamicUser now).

Also, "All supported distributions now use the same system user names" is confusing -- they *always* did, that didn't change. We removed the configurability because in practice it was the same everywhere anyway.

Reviewer 2: @garrett Thanks! I committed your other suggestions, and updated the sysusers text. I took most things from your suggestion, but reordered it a bit, made it more succinct, and fixed the above issues.

> ## Metrics: Prefer valkey over redis for Grafana setup

Reviewer: ```suggestion
## Metrics: Grafana setup now prefers Valkey over Redis
```

We release regularly.

Here are the release notes from Cockpit 317:

## webserver: System user changes

Cockpit has changed how the system user is handled. All supported distributions use the same system user names. We don't test, support, or recommend running the web server as root, which had previously been the default with an upstream build and install. Hence, the following `./configure` options related to static users have been removed:
 - `--with-cockpit-user`
 - `--with-cockpit-group`
 - `--with-cockpit-ws-instance-user`
 - `--with-cockpit-ws-instance-group`

The `cockpit-ws` system user is no longer statically created, but created transiently and on-demand via [systemd's `DynamicUser` feature](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=). If you would like to remove the previously-used static system user:
 1. Upgrade to Cockpit 317 or later
 2. Run `systemctl stop cockpit`
 3. Run `userdel -r cockpit-ws`

The `cockpit-wsinstance` system user is now declared through [systemd-sysusers](https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html#).

## Metrics: Grafana setup now prefers Valkey over Redis

> The [Grafana metrics setup](https://cockpit-project.org/blog/pcp-grafana.html) ("Export to network") now prefers using [valkey](https://valkey.io/) over [redis](https://redis.io/) on Fedora. See the [Fedora change proposal](https://fedoraproject.org/wiki/Changes/Replace_Redis_With_Valkey) for details.

Reviewer 3: The Grafana metrics setup ("Export to network") now prefers... scans weird.

It would work fine without the ("Export to network") or if it was ("Expect to network" panel) or so.

Reviewer 2: I dropped the "Export to network", that makes it a bit shorter.
